### DISCUSSION TITLE
Int format

### DISCUSSION THREAD

**Discussion by taichi-ishitani** [apple/pkl#577] [Q&A] [answered]
I have a question about Int format.

Following Pkl code can be parsed as Int.

```pkl
a = 1_2__3__
```
![image](https://github.com/user-attachments/assets/a69065dc-1317-453f-911b-b09b8b9399ec)

But an error will be reported for the Pkl code below.

```pkl
a = "1_2__3__".toInt()
```
![image](https://github.com/user-attachments/assets/05819ba3-a623-44ea-9e5d-0744064a2ac3)

Which behavior is correct?

**Comment by bioball** [comment, marked as answer]
`1_2__3__` is a valid int literal, with underscore separators. It probably makes sense for `String#toInt()` to be able to parse these values.

**Reply by taichi-ishitani** [reply]
OK, I understood.
I will create a PR to allow `String#toInt()` to parse a Int string including `__`.
